File: src/test/test_type/6.for/test0.pas
program TestForLoop(input, output);
var
  i, start, endInt: integer;
  startReal, endReal: real;
arr3: array [1..5] of real;
begin
  start := 1;
  endInt := 5;

  // 正确的for循环
  for i := start to endInt do
  begin
    write(i);
  end;

  // 错误的for循环：开始值为实数类型
  startReal := 1.5;
  for arr3 := startReal to endInt do // 类型错误：开始值为实数类型
  begin
    write(i);
  end;
  // 错误的for循环：结束值为实数类型
  endReal := 5.5;
  for i := start to endReal do // 类型错误：结束值为实数类型
  begin
    write(i);
  end;
  // 错误的for循环：开始值和结束值都是实数类型
  for i := startReal to endReal do // 类型错误：开始值和结束值都是实数类型
  begin
    write(i);
  end;
end.

File: test/test_type/6.for/c_code/test0.c
#include <stdio.h>

float arr3[5-1+1];
float startreal,endreal;
int i,start,endint;

int main()
{
	start = 1;
	endint = 5;
	for(i = start; i <= endint; i ++)
	{
		printf("%d", i);
	}
	startreal = 1.5;
	for(arr3 = startreal; arr3 <= endint; arr3 ++)
	{
		printf("%d", i);
	}
	endreal = 5.5;
	for(i = start; i <= endreal; i ++)
	{
		printf("%d", i);
	}
	for(i = startreal; i <= endreal; i ++)
	{
		printf("%d", i);
	}
	return 0;
}